cmd/kube-scheduler/app/server.go

```diff
@@ -169,11 +169,10 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 	logger.Info("Golang settings", "GOGC", os.Getenv("GOGC"), "GOMAXPROCS", os.Getenv("GOMAXPROCS"), "GOTRACEBACK", os.Getenv("GOTRACEBACK"))
 
 	// Configz registration.
-	if cz, err := configz.New("componentconfig"); err == nil {
-		cz.Set(cc.ComponentConfig)
-	} else {
+	if cz, err := configz.New("componentconfig"); err != nil {
 		return fmt.Errorf("unable to register configz: %s", err)
 	}
+	cz.Set(cc.ComponentConfig)
```

Comment:
this still needs to be in an `else` - the lifetime of `cz` is the `if` clause.

Reply:
Thank you for the feedback! I’ve updated the code to use an else clause, ensuring the scope of cz remains within the if block. Please let me know if there’s anything further to adjust.